Dockerfile

```diff
@@ -1,31 +1,33 @@
-FROM python:3.6-slim
+FROM python:3.6.9-slim
 
-# add a non-privileged user for installing and running
-# the application
-RUN groupadd -g 10001 app && \
-    useradd -d /app -g 10001 -G app -M -s /bin/sh -u 10001 app
+# Set up user and group.
+ARG groupid=10001
+ARG userid=10001
 
 WORKDIR /app
+RUN groupadd --gid $groupid app && \
+    useradd -g app --uid $userid --shell /usr/sbin/nologin --create-home app
 
-# Open a shell by default.
-ENTRYPOINT ["/app/conf/run.sh"]
+# Set entrypoint for this image. The entrypoint script takes a service
+# to run as the first argument. See the script for available arguments.
+ENTRYPOINT ["/app/docker/app_entrypoint.sh"]
 CMD ["shell"]
 
 # Create an app user owned var/run section.
 RUN mkdir -p /var/run/location/ && chown -R app:app /var/run/location/
 
-# Disable installing doc/man/locale files
+# Disable installing doc/man/locale files.
 RUN echo "\
 path-exclude=/usr/share/doc/*\n\
 path-exclude=/usr/share/man/*\n\
 path-exclude=/usr/share/locale/*\n\
 " > /etc/dpkg/dpkg.cfg.d/apt-no-docs
 
-# Install GNUPG
+# Install GNUPG.
 RUN apt-get update && apt-get -y install gnupg
 
-# Add MySQL apt repo & GPG key
-COPY ./conf/mysql_pubkey.asc /tmp/mysql_pubkey.asc
+# Add MySQL apt repo & GPG key.
+COPY ./docker/mysql_pubkey.asc /tmp/mysql_pubkey.asc
```

Comment:
I moved all things related to building images into the `docker/` subdirectory including this pubkey.